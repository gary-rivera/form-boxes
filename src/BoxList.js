import React, { useState } from 'react';
import Box from './Box';
import BoxForm from './NewBoxForm';
import { v4 as uuid } from 'uuid';

function BoxList() {
  const [boxes, setBoxes] = useState([]);

  function renderBox(boxData) {
    let newBox = { ...boxData, id: uuid() };
    setBoxes(boxes => [...boxes, newBox]);
  }

  function removeBox(id) {
    setBoxes(oldBoxes => oldBoxes.filter(box => box.id !== id))
  }

  return (
    <div>
      <BoxForm create={renderBox} />
      <div>

        {boxes.map(({ height, width, color, id }) => (
          <Box
            key={id}
            height={`${height}px`}
            width={`${width}px`}
            backgroundColor={color}
            removeBox={() => removeBox(id)}
          />
        ))}

      </div>
    </div>)
}

export default BoxList;