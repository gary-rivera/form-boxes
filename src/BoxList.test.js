import { fireEvent, render } from '@testing-library/react'
import BoxList from './BoxList'

it("renders without crashing", function () {
    render(<BoxList />)
})

it("matches default snapshot", function () {
    const { container } = render(<BoxList />)
    expect(container).toMatchSnapshot();
})

it("creates boxes", function() {
    const {container, getByLabelText, queryByText, debug} = render(<BoxList />)

    const heightInput = getByLabelText("Height");
    const widthInput = getByLabelText("Width");
    const colorInput = getByLabelText("Color");
    const submitBtn = queryByText("Add box")

    fireEvent.change(heightInput, {target: {value:'80'}});
    fireEvent.change(widthInput, {target: {value:'75'}});
    fireEvent.change(colorInput, {target: {value:'black'}});
    fireEvent.click(submitBtn);

    const box = container.querySelector(".Box");
    expect(box).toBeInTheDocument();

    const removeBtn = queryByText("Remove The Box!");
    fireEvent.click(removeBtn);
    expect(removeBtn).not.toBeInTheDocument();
    expect(box).not.toBeInTheDocument();

    expect(container).toMatchSnapshot();
})