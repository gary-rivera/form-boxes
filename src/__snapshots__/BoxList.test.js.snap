// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`creates boxes 1`] = `
<div>
  <div>
    <div>
      <form>
        <label
          for="height"
        >
          Height
        </label>
        <input
          id="height"
          name="height"
          value=""
        />
        <label
          for="width"
        >
          Width
        </label>
        <input
          id="width"
          name="width"
          value=""
        />
        <label
          for="color"
        >
          Color
        </label>
        <input
          id="color"
          name="color"
          value=""
        />
        <button>
          Add box
        </button>
      </form>
    </div>
    <div />
  </div>
</div>
`;

exports[`matches default snapshot 1`] = `
<div>
  <div>
    <div>
      <form>
        <label
          for="height"
        >
          Height
        </label>
        <input
          id="height"
          name="height"
          value=""
        />
        <label
          for="width"
        >
          Width
        </label>
        <input
          id="width"
          name="width"
          value=""
        />
        <label
          for="color"
        >
          Color
        </label>
        <input
          id="color"
          name="color"
          value=""
        />
        <button>
          Add box
        </button>
      </form>
    </div>
    <div />
  </div>
</div>
`;
